crest/Assets/Crest/Crest/Scripts/LodData/LodDataMgrShadow.cs

```diff
@@ -184,29 +184,27 @@ public override void UpdateLodData()
                 var lt = OceanRenderer.Instance._lods[lodIdx];
 
                 lt._renderData.Validate(0, this);
-                _renderMaterial[lodIdx].SetVector("_CenterPos", lt._renderData._posSnapped);
-                _renderMaterial[lodIdx].SetVector("_Scale", lt.transform.lossyScale);
-                _renderMaterial[lodIdx].SetVector("_CamPos", OceanRenderer.Instance.Viewpoint.position);
-                _renderMaterial[lodIdx].SetVector("_CamForward", OceanRenderer.Instance.Viewpoint.forward);
-                _renderMaterial[lodIdx].SetVector("_JitterDiameters_CurrentFrameWeights", new Vector4(Settings._jitterDiameterSoft, Settings._jitterDiameterHard, Settings._currentFrameWeightSoft, Settings._currentFrameWeightHard));
-                _renderMaterial[lodIdx].SetMatrix("_MainCameraProjectionMatrix", _cameraMain.projectionMatrix * _cameraMain.worldToCameraMatrix);
-                _renderMaterial[lodIdx].SetFloat("_SimDeltaTime", Time.deltaTime);
+                _renderMaterial[lodIdx].SetVector(Shader.PropertyToID("_CenterPos"), lt._renderData._posSnapped);
```

Comment:
@huwb This was the bit that I was most unsure about, because on one hand forcing ourselves to do this explictly is a good reminder that we should be creating property IDs for all shader params up-front. However, it is a bit clunky when done inline like this. I'm happy to implement this conversion inside the property wrapper if you think that is preferable.

Reply:
Thanks Tom. This was lazyness on my part - this class should store the IDs as member variables. Also clunky but that's the system. I'll see if I can point to an example..

Reply:
Didn't find a perfect simple example. But as I said it could store the IDs as static int member variables and initialise them on construction or init. Let me know if that makes sense.

Reply:
👍 Will do that now